bundles/org.openhab.binding.lghombot/src/main/java/org/openhab/binding/lghombot/internal/LGHomBotHandler.java

```diff
@@ -0,0 +1,722 @@
+        }
+        logger.debug("Status received: {}", status);
+    }
+
+    private void refreshFromState(ChannelUID channelUID) {
+        switch (channelUID.getId()) {
+            case CHANNEL_STATE:
+                updateState(channelUID, StringType.valueOf(currentState));
+                break;
+            case CHANNEL_CLEAN:
+                updateState(channelUID, currentCleanState);
+                break;
+            case CHANNEL_START:
+                updateState(channelUID, currentStartState);
+                break;
+            case CHANNEL_HOME:
+                updateState(channelUID, currentHomeState);
+                break;
+            case CHANNEL_BATTERY:
+                updateState(channelUID, currentBattery);
+                break;
+            case CHANNEL_CPU_LOAD:
+                updateState(channelUID, currentCPULoad);
+                break;
+            case CHANNEL_SRV_MEM:
+                updateState(channelUID, StringType.valueOf(currentSrvMem));
+                break;
+            case CHANNEL_TURBO:
+                updateState(channelUID, currentTurbo);
+                break;
+            case CHANNEL_REPEAT:
+                updateState(channelUID, currentRepeat);
+                break;
+            case CHANNEL_MODE:
+                updateState(channelUID, StringType.valueOf(currentMode));
+                break;
+            case CHANNEL_NICKNAME:
+                updateState(channelUID, StringType.valueOf(currentNickname));
+                break;
+            case CHANNEL_CAMERA:
+                parseImage();
+                updateState(channelUID, currentImage);
+                break;
+            case CHANNEL_LAST_CLEAN:
+                updateState(channelUID, currentLastClean);
+                break;
+            case CHANNEL_MAP:
+                parseMap();
+                updateState(channelUID, currentMap);
+                break;
+            case CHANNEL_MONDAY:
+            case CHANNEL_TUESDAY:
+            case CHANNEL_WEDNESDAY:
+            case CHANNEL_THURSDAY:
+            case CHANNEL_FRIDAY:
+            case CHANNEL_SATURDAY:
+            case CHANNEL_SUNDAY:
+                refreshCounter = 1;
+                break;
+            default:
+                logger.warn("Channel refresh for {} not implemented!", channelUID.getId());
+        }
+    }
+
+    private void updateAllChannels() {
+        if (disposed) {
+            return;
+        }
+        if (refreshCounter > 0) {
+            refreshCounter--;
+            if (refreshCounter == 0) {
+                fetchSchedule();
+                return;
+            }
+            if (refreshCounter % 5 == 1) {
+                parseImage();
+                updateState(new ChannelUID(getThing().getUID(), CHANNEL_CAMERA), currentImage);
+                return;
+            }
+        }
+        String status = null;
```

Comment:
I have seen this next 8 lines before, please re-use that code. If `sendCommand `would return the status and you would call `sendCommand("/status.txt")` here you would be done.